### PR TITLE
fix grpc supported version

### PR DESCRIPTION
**Pull request by JaredTan95**
# Description
Our python application uses `grpcio==1.34.0` (greater than version `1.27` as described in the documentation)

The `opentelemetry-instrumentation-grpc 1.27.0` plugin conflicts with `grpcio==1.34.0` , resulting in the following error:

```txt
/olet-auto-instrumentation-python/opentetelemetry/instrumerflation/grpe/__inft__:ру , timhe 402, In server
return self._original_func(*args, **kwargs)
tile "/usr/local/lib/python3.9/site-packages/grpc/aio/server.ny", line 206, in server
return Server(migration_thread_pool, () if handlersis is None else handlers,
File "/usr/local/lib/python3.9/site-packages/grpc/aio/_server.py", line 53, in __init__
self._server = cygrpc.AioServer(
File "src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi", line 813, in grpc._cython.cygrpc.AioServer.__init__
TypeError: Expected tuple, got list
```

This has been fixed for grpc in version `1.42.0` 3 years ago: https://github.com/grpc/grpc/commit/96995801d878b4d39f329a51d664902e6125c07a

### CODE REVIEW

File: instrumentation/README.md [hunk 1]
```diff
@@ -21,7 +21,7 @@
 | [opentelemetry-instrumentation-falcon](./opentelemetry-instrumentation-falcon) | falcon >= 1.4.1, < 4.0.0 | Yes | experimental
 | [opentelemetry-instrumentation-fastapi](./opentelemetry-instrumentation-fastapi) | fastapi ~= 0.58 | Yes | migration
 | [opentelemetry-instrumentation-flask](./opentelemetry-instrumentation-flask) | flask >= 1.0 | Yes | migration
-| [opentelemetry-instrumentation-grpc](./opentelemetry-instrumentation-grpc) | grpcio ~= 1.27 | No | experimental
+| [opentelemetry-instrumentation-grpc](./opentelemetry-instrumentation-grpc) | grpcio >=1.42.0 | No | experimental
```
Comment (xrmx) [inline instrumentation/README.md:24]:
```suggestion
| [opentelemetry-instrumentation-grpc](./opentelemetry-instrumentation-grpc) | grpcio >= 1.42.0 | No | experimental
```

**Comment by JaredTan95** [comment]
@xrmx can you help add an `skipchanlog` label?

**Comment by xrmx** [comment]
I think you need to add the changelog and also change the code in `instrumentation/opentelemetry-instrumentation-grpc/package.py` to match documentation and then run `python scripts/generate_instrumentation_bootstrap.py` to regenerate `bootstrap_gen.py`

**Review by xrmx — CHANGES_REQUESTED** [review]
I think you need to add the changelog and also change the code in `instrumentation/opentelemetry-instrumentation-grpc/package.py` to match documentation and then run `python scripts/generate_instrumentation_bootstrap.py` to regenerate `bootstrap_gen.py`

**Comment by JaredTan95** [comment]
```bash
python3 scripts/generate_instrumentation_bootstrap.py
>>> black --config github/opentelemetry-python-contrib/pyproject.toml .
reformatted github/opentelemetry-python-contrib/opentelemetry-instrumentation/src/opentelemetry/instrumentation/bootstrap_gen.py

All done! ✨ 🍰 ✨
1 file reformatted, 12 files left unchanged.
>>> isort --settings-path github/opentelemetry-python-contrib/.isort.cfg --profile black .
INFO:instrumentation_list_generator:generatedgithub/opentelemetry-python-contrib/opentelemetry-instrumentation/src/opentelemetry/instrumentation/bootstrap_gen.py
```

There seems to be no changes in `bootstrap_gen.py`

**Comment by JaredTan95** [comment]
> Yeah to the list of things to update I forgot the package `pyproject.toml`

thx, done~

**Review by emdneto — COMMENTED** [review]
And also at the end run `tox -e generate`

**Review by xrmx — APPROVED** [review]

**Review by lzchen — APPROVED** [review]

**Pull request merged**